docker-compose.yml

```diff
@@ -88,7 +88,7 @@ services:
             - "127.0.0.1:7004:7004"
             - "127.0.0.1:7005:7005"
     mongo:
-        image: mongo:3.6
+        image: mongo:4.4
```

Comment:
I was going to suggest also updating the version used in gitlab, but I assume it's new enough that it works for this use case https://github.com/DataDog/dd-trace-py/blob/caefec1844cdb0c4ed15727af8f1a742711bf0e2/.gitlab/services.yml#L27